UI/GovernmentWnd.cpp

```diff
@@ -229,35 +238,56 @@ void PolicyControl::CompleteConstruction() {
     GG::Control::CompleteConstruction();
     if (!m_policy)
         return;
+    int empire_id = GGHumanClientApp::GetApp()->EmpireID();
+    const ScriptingContext context;
+    const auto& name = UserString(m_policy->Name());
+    const auto& cost = static_cast<int>(m_policy->AdoptionCost(empire_id, context));
+
     //std::cout << "PolicyControl: " << m_policy->Name() << std::endl;
 
     m_background = GG::Wnd::Create<GG::StaticGraphic>(PolicyBackgroundTexture(m_policy),
                                                       GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE);
-    m_background->Resize(GG::Pt(SLOT_CONTROL_WIDTH, SLOT_CONTROL_HEIGHT));
+    m_icon = GG::Wnd::Create<GG::StaticGraphic>(ClientUI::PolicyIcon(m_policy->Name()),
+                                                      GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE);
+    m_name_label = GG::Wnd::Create<GG::TextControl>(GG::X0, GG::Y0, GG::X1, GG::Y1, name,
+                                                      ClientUI::GetBoldFont(), ClientUI::TextColor(), GG::FORMAT_WORDBREAK);
+    m_cost_label = GG::Wnd::Create<GG::TextControl>(GG::X0, GG::Y0, GG::X1, GG::Y1, std::to_string(cost),
+                                                      ClientUI::GetBoldFont(), ClientUI::TextColor());
+
     m_background->Show();
-    AttachChild(m_background);
+    m_icon->Show();
 
+    AttachChild(m_background);
+    AttachChild(m_icon);
+    AttachChild(m_name_label);
+    AttachChild(m_cost_label);
 
-    // position of policy image centred within policy control.  control is size of a slot, but the
-    // policy image is smaller
-    GG::X policy_left = (Width() - POLICY_CONTROL_WIDTH) / 2;
-    GG::Y policy_top = (Height() - POLICY_CONTROL_HEIGHT) / 2;
+    Resize(Size());
 
     //DebugLogger() << "PolicyControl::PolicyControl this: " << this << " policy: " << policy << " named: " << (policy ? policy->Name() : "no policy");
-    m_icon = GG::Wnd::Create<GG::StaticGraphic>(ClientUI::PolicyIcon(m_policy->Name()),
-                                                GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE);
-    m_icon->MoveTo(GG::Pt(policy_left, policy_top));
-    m_icon->Resize(GG::Pt(POLICY_CONTROL_WIDTH, POLICY_CONTROL_HEIGHT));
-    m_icon->Show();
-    AttachChild(m_icon);
-
     SetDragDropDataType(POLICY_CONTROL_DROP_TYPE_STRING);
 
     //DebugLogger() << "PolicyControl::PolicyControl policy name: " << m_policy->Name();
     SetBrowseModeTime(GetOptionsDB().Get<int>("ui.tooltip.delay"));
     SetBrowseInfoWnd(PolicyBrowseWnd(m_policy->Name()));
 }
 
+void PolicyControl::Resize(const GG::Pt& sz, const int pts) {
+    m_background->Resize(sz);
+    m_icon->Resize(GG::Pt(sz.x, sz.y * 2/3));
+
+    std::shared_ptr<GG::Font> font = ClientUI::GetBoldFont(pts);
+
+    m_name_label->SetFont(font);
+    m_cost_label->SetFont(font);
```

Comment:
```suggestion
    m_cost_label->SetFont(std::move(font));
```
if you don't use the poniter again, can move it